Sprint Six - Work on the functionality of the sprint six terms of ref…

vil-shavetail opened the pull request.

vil-shavetail committed:
Sprint Six - Work on the functionality of the sprint six terms of reference.

vil-shavetail committed:
Sprint Six - Fixed errors that were found by automatic error checking github system.

vil-shavetail committed:
Sprint Six - Fixed errors that were found by automatic error checking github system in InMemoryHistoryManager class.

vil-shavetail committed:
Sprint Six - Fixed errors that were found by automatic error checking github system in InMemoryTaskManager class.

vil-shavetail committed:
Sprint Six - Fixed errors that were found by automatic error checking github system in InMemoryTaskManager class.

vil-shavetail committed:
Sprint Six - Fixed errors that were found by automatic error checking github system in Main class.

KirPyt commented on .idea/misc.xml:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_21" project-jdk-name="openjdk-21" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_21" project-jdk-name="liberica-21" project-jdk-type="JavaSDK">
```
Данный файл представляет собой локальную конфигурацию Вашей среды, отправлять его на проверку (включать в коммит) не нужно

vil-shavetail replied:
Данное замечание исправил.

KirPyt commented on src/ru/yandex/practicum/managers/InMemoryHistoryManager.java:
```diff
@@ -3,24 +3,94 @@
+            Node node = new Node();
+            node.setTask(task);
+
+            if (data.containsKey(task.getId())) {
```
Рекомендую разделить логику добавления объекта в список и проверок, обусловленных Вашей бизнес логикой. В данном случае - добавление в список, это функционал самого списка, а вот удаление повторных объектов - это требования к Вашему приложению

vil-shavetail replied:
Вот это замечание не понял.

KirPyt replied:
В данном замечании имеется в виду, что в методе add перед добавлением элемента в список Вы будете проверять его наличие и удалять

Соотвественно из метода link надо удалить такую проверку, и оставить в методе link только добавление узла в список

KirPyt commented on src/ru/yandex/practicum/managers/InMemoryHistoryManager.java:
```diff
@@ -3,24 +3,94 @@
 
-    private static final int MAX_HISTORY_SIZE = 10;
-    private final List<Task> history = new ArrayList<>();
+    private static class HandMadeHistoryLinkedList {
```
Рекомендую вынести класс связанного списка в отдельный файл, чтобы была возможность поддерживать его отдельно от логики менеджера истории

vil-shavetail replied:
Вынес в отдельный файл.

KirPyt commented on src/ru/yandex/practicum/managers/InMemoryHistoryManager.java:
```diff
@@ -3,24 +3,94 @@
+            }
+        }
+
+        private Node getNode(int id) {
```
Не рекомендую давать доступ к внутренней логике списка, так как может привести к его модификации в обход задуманной логики

vil-shavetail replied:
Метод использую для того, чтобы в последствии удалить узел в списке. Пока не придумал как без него работать.

KirPyt replied:
Узел в списке удаляется через метод remove. Стоит постараться использовать только его так как он позволяет удалить данные правильно. Иные способы могут усложнить логику

KirPyt commented on src/ru/yandex/practicum/managers/InMemoryHistoryManager.java:
```diff
@@ -3,24 +3,94 @@
+    private final HandMadeHistoryLinkedList handMadeHistoryLinkedList = new HandMadeHistoryLinkedList();
 
     @Override
     public void add(Task task) {
```
Собственно в данном методе как раз будет проверка и удаление дубликата, а затем добавление в связанный список

vil-shavetail replied:
Видимо, связано вот с этим замечанием https://github.com/vil-shavetail/java-kanban/pull/1#discussion_r1909358305
Пока не пойму что нужно сделать.

KirPyt commented on src/ru/yandex/practicum/managers/Node.java:
```diff
@@ -0,0 +1,33 @@
+
+import ru.yandex.practicum.tasks.Task;
+
+public class Node {
```
Рекомендую сделать данный класс внутренним классом Вашего связанного списка, так как он относится к внутренней структуре связанного списка

vil-shavetail replied:
Сделал внутренним классом списка.

KirPyt reviewed (commented):
Здравствуйте!

Критических замечаний по работе нет, но есть некоторые рекомендации, на которые стоит обратить внимание, они позволят Вам сделать логику работы со списком более прозрачной и легкой в поддержке

vil-shavetail committed:
Sprint Six - Fixed the reviewer's remarks after first review.
Fixed:
1) https://github.com/vil-shavetail/java-kanban/pull/1#discussion_r1909356396
2) https://github.com/vil-shavetail/java-kanban/pull/1#discussion_r1909361383
3) https://github.com/vil-shavetail/java-kanban/pull/1#discussion_r1909362262

KirPyt commented:
Здравствуйте!

Ответил в комментариях по поводу прошлых рекомендаций. В целом критических замечаний нету, Ваша работа зачтена
